src/thunderbots/backend_output/grsim/grsim_backend.cpp

```diff
@@ -1,11 +1,54 @@
-#include "grsim_backend.h"
+#include "backend_output/grsim/grsim_backend.h"
+#include <iostream>
+#include <optional>
 #include "ai/primitive/move_prim.h"
+#include "proto/grSim_Commands.pb.h"
+#include "proto/grSim_Packet.pb.h"
 
-GrSimBackend::GrSimBackend()
+using namespace boost::asio;
+
+GrSimBackend::GrSimBackend(std::string network_address, unsigned short port)
+    : network_address(network_address), port(port), socket(io_service)
+{
+    socket.open(ip::udp::v4());
+    remote_endpoint = ip::udp::endpoint(ip::address::from_string(network_address), port);
+}
+
+GrSimBackend::~GrSimBackend()
+{
+    socket.close();
+}
+
+void GrSimBackend::setRobotVelocities(
+    unsigned int robot_id, bool team_colour_yellow, Point velocity,
+    Angle angular_velocity)
 {
+    grSim_Packet packet;
+
+    packet.mutable_commands()->set_isteamyellow(team_colour_yellow);
+    packet.mutable_commands()->set_timestamp(0.0);
+    grSim_Robot_Command* command = packet.mutable_commands()->add_robot_commands();
+
+    command->set_id(robot_id);
+
+    // We set a robot velocity, not individual wheel velocities
+    command->set_wheelsspeed(false);
+    command->set_veltangent(static_cast<float>(velocity.x()));
```

Comment:
Space here would be helpful to make it clear this comment has nothing to do with setting velocity, ie:
```
   // We set a robot velocity, not individual wheel velocities
   command->set_wheelsspeed(false);
   
   command->set_veltangent(static_cast<float>(velocity.x()));
```